### PR TITLE
Use VALIDATE macro to validate WML in teleport.cpp (#8175)

### PR DESCRIPTION
**Pull request by dsfb**
This code is an implementation that can solve the issue #8175.
(Sorry, but it is not tested!)

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
I'll run CI to make sure it actually compiles. Besides that, I think the error should say `[tunnel] tag` instead of just `config`. Someone seeing that error wouldn't understand what is mean by "config".

**Review by CelticMinstrel — CHANGES_REQUESTED** [review]
See my previous comment.

**Comment by CelticMinstrel** [comment]
…

That's equally wrong.

Though using that function to generate the message wasn't a bad idea, it doesn't help if the values you pass to it are incorrect. `"cfg"` is a C++ implementation detail that will be meaningless to anyone who actually sees this error message. The correct first parameter is `"tunnel"`.

Also, you need make the same changes in all the VALIDATE calls you added here. Your last change updated the first two but left the rest untouched.

**Comment by gfgtdf** [comment]
imo the [filter] tag should be optional but we can also do this after this was merged.

Also we are a in a string freeze, maybe it'd be better not to makr these strings as translatable, and to mark the translation mark later.

Lastly, i don't know how the VALIDATE macros behave (in any case this is certainly an improvement and should be merged), the default behavior for wrong wml syntax is just to print an error message and to go on, if validate causes the game to be aborted this pr would be a change from the usual behavior.

**Comment by CelticMinstrel** [comment]
I believe VALIDATE macros cause a WML error dialog to be shown and dump you to the title screen. At least, that should be true if triggered during loading. I'm not sure what would happen if this is triggered from ActionWML. It might be the same but I'm not sure.

File: src/pathfind/teleport.cpp
Comment (stevecotton) [inline src/pathfind/teleport.cpp:53]: I'd support adding an equivalent of `missing_mandatory_wml_key` which says "In section ‘[$section|]’ the key ‘$key|’ is set more than once." along with a `VALIDATE_EXACTLY_ONE_CHILD` macro which uses `missing_mandatory_wml_key` or the new message as appropriate.

Comment (CelticMinstrel) [reply src/pathfind/teleport.cpp:53]: It's still better than assert though.

**Comment by dsfb** [comment]
Hi, people!

I have pushed another commit to fixing the problems that you have reported.
Do you know if the previous problems are solved now, or not?

Thanks.

File: src/pathfind/teleport.cpp
Comment (Pentarctagon) [inline src/pathfind/teleport.cpp:51]: Per gfgtdf's comment:
> Also we are a in a string freeze, maybe it'd be better not to makr these strings as translatable, and to mark the translation mark later.

these should be left as untranslatable strings for now.

**Comment by Pentarctagon** [comment]
I think otherwise this can be merged at least as an improvement, if maybe not the final, ideal state?

**Comment by CelticMinstrel** [comment]
> Do you know if the previous problems are solved now, or not?

You still haven't addressed my original comment.

**Comment by dsfb** [comment]
> > Do you know if the previous problems are solved now, or not?
> 
> You still haven't addressed my original comment.
> 
> > I think otherwise this can be merged at least as an improvement, if maybe not the final, ideal state?
> 
> Only if the message is changed to be correct. "config" is decidedly not correct.

@CelticMinstrel I have made another commit. What do you think about it?

**Review by CelticMinstrel — APPROVED** [review]
It's not quite what I envisioned, but it's close enough – it gives sufficient information for someone seeing the message to find where the error came from.

File: src/pathfind/teleport.cpp
Comment (stevecotton) [reply src/pathfind/teleport.cpp]: Still an issue with commit 6d2839c

**Comment by dsfb** [comment]
> It's not quite what I envisioned, but it's close enough – it gives sufficient information for someone seeing the message to find where the error came from.

Thanks, @CelticMinstrel!

**Review by stevecotton — CHANGES_REQUESTED** [review]
It's still adding translatable strings, so no expecting to merge tonight.

**Comment by Pentarctagon** [comment]
@dsfb are you going to be removing the translation mark from the new strings?

**Comment by CelticMinstrel** [comment]
If not, we'll probably merge this about one month from now, after 1.18 is branched of.

**Comment by dsfb** [comment]
@Pentarctagon 
@CelticMinstrel 
I made another commit for removing the translation mark from the new strings in the teleport code.

**Comment by stevecotton** [comment]
Looks good to me, just waiting for the CI to finish

**Pull request merged**
